xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -386,7 +386,7 @@ bool CDVDDemuxFFmpeg::Open(CDVDInputStream* pInput, bool streaminfo, bool filein
 
     m_pFormatContext->pb = m_ioContext;
 
-    AVDictionary *options = NULL;
+    AVDictionary *options = GetFFMpegOptionsFromInput();
```

Comment:
I can give a log too if you want but PR is not the good place for that.
Nevermind, you are right, In fact kodi use the extension to "choose its method", if I rename the playlist file with the .m3u8 extension all will works fine.

I m not skilled enough to check why mimetype doesn't work, but it's another problem, so you can set this PR to invalid.

But I have a last question, I haven't finished to read all the code tracking, but where is the problem to use the ffmpeg option with custom io handling ? If I put a for exemple a cookie in command option (with kodi api), it's to use it in all cases, whatever the engine the core will use.